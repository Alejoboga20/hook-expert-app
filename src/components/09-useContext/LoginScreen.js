import React, { useContext } from 'react';
import { UserContext } from './UserContext';

const LoginScreen = () => {
  const { setUser } = useContext(UserContext);

  const loginUser = {
    name: 'Alejo',
    email: 'alejo@email.com'
  };

  return (
    <div>
      <h1>Login Screen</h1>
      <hr />
      <button className='btn btn-primary' onClick={() => setUser(loginUser)}>
        Login
      </button>
    </div>
  );
};

export default LoginScreen;
