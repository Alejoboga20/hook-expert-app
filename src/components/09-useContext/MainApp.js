import React from 'react';
import AppRouter from './AppRouter';
import { UserContext } from './UserContext';

const MainApp = () => {
  const user = {
    id: 1234,
    name: 'Alejo',
    email: 'test@gmail.com'
  };

  return (
    <div>
      <UserContext.Provider value={user}>
        <AppRouter />
      </UserContext.Provider>
    </div>
  );
};

export default MainApp;
