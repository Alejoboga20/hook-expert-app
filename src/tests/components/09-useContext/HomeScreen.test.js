import { mount } from 'enzyme';
import React from 'react';
import HomeScreen from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('HomeScreen tests', () => {
  const user = {
    name: 'test',
    email: 'test@email.com'
  };

  const wrapper = mount(
    <UserContext.Provider value={{ user }}>
      <HomeScreen />
    </UserContext.Provider>
  );

  test('should display properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
