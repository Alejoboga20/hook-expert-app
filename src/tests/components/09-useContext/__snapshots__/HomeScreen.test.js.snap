// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen tests should display properly 1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <hr />
  <pre>
    {
   "name": "test",
   "email": "test@email.com"
}
  </pre>
</div>
`;
